services/placement/algo/sharded.go

```diff
@@ -157,10 +158,13 @@ func (a rackAwarePlacementAlgorithm) ReplaceInstance(
 }
 
 func (a rackAwarePlacementAlgorithm) addInstance(p services.ServicePlacement, addingInstance services.PlacementInstance) (services.ServicePlacement, error) {
-	if _, exist := p.Instance(addingInstance.ID()); exist {
-		return nil, errAddingInstanceAlreadyExist
-
+	if instance, exist := p.Instance(addingInstance.ID()); exist {
```

Comment:
Sometimes when the users tries to add an instance, the instance will have correct rack,zone,weight info but not neccessarily the correct shards info, using the instance found from the placement gets the correct shards data